colossalai/tensor/colo_tensor.py

```diff
@@ -103,7 +104,7 @@ def init_from_torch_tensor(tensor: torch.Tensor, save_payload=True, is_model_dat
                             requires_grad=tensor.requires_grad,
                             pin_memory=tensor.is_pinned(),
                             device=tensor.device,
-                            torch_tensor=tensor if save_payload else torch.empty(0),
+                            torch_tensor=tensor.detach() if save_payload else torch.empty(0),
```

Comment:
why add detach() here?

Reply:
optimizer need all parameters as a leaf node in the graph.
without detach() will not pass the ColoTensor to an optimizer

Reply:
Polish it in a better way.